api/validator.js

```diff
@@ -36,7 +36,7 @@ const SCHEMA_JOB_PERMUTATION = Joi.object()
         secrets: Job.secrets,
         settings: Job.settings,
         sourcePaths: Job.sourcePaths,
-        stages: Base.stages,
+        stage: Base.stage,
```

Comment:
Do we need to allow the entire stage object to be stored in a job permutation?
```
const SCHEMA_STAGE = Joi.object()
    .keys({
        description: Joi.string(),
        jobs: Joi.array().items(Job.jobName).min(0),
        setup: SCHEMA_SETUP_TEARDOWN_JOB,
        teardown: SCHEMA_SETUP_TEARDOWN_JOB,
        requires: Job.requires
    })
    .unknown(false);
```

I believe we only need stage `name` (or `id`) for reverse lookup per [this change](https://github.com/screwdriver-cd/config-parser/pull/143/files#diff-650a22c636f250a0b20166054e7ffaa7a337afe42b97be42acc2728c96107d0aR564).
```suggestion
        stage: Joi.object({
            name:  Joi.string().regex(Regex.STAGE_NAME).required()
        }).keys().unknown(false),
```